### PR TITLE
OS Modifier (EMU): Add sshkeys to user config

### PR DESCRIPTION
**Pull request by elainezhao96**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Change Log  
https://github.com/microsoft/CBL-Mariner/pull/6824#discussion_r1406527775
as recommended, this is a follow-up separate pt for previous emu pr, 

###### Does this affect the toolchain?  
NO

### CODE REVIEW

File: toolkit/tools/imagegen/configuration/user.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ type User struct {
 	Password            string   `json:"Password"`
 	PasswordExpiresDays int64    `json:"PasswordExpiresDays"`
 	SSHPubKeyPaths      []string `json:"SSHPubKeyPaths"`
+	SSHPubKeys          []string `yaml:"SSHPubKeys"`
```
Comment (cwize1) [inline toolkit/tools/imagegen/configuration/user.go:24]: Change `yaml` to `json`.

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 2]
```diff
@@ -1550,6 +1550,18 @@ func ProvisionUserSSHCerts(installChroot safechroot.ChrootInterface, username st
...
+		logger.Log.Infof("Adding ssh key (%s) to user (%s) .ssh/authorized_users", filepath.Base(pubKey), username)
+
+		// Directly append the key string with a newline
+		pubKey += "\n"
+		err = file.Append(pubKey, authorizedKeysTempFile)
+		if err != nil {
+			logger.Log.Warnf("Failed to append to %s : %v", authorizedKeysTempFile, err)
+			return
+		}
+	}
```
Comment (romoh) [inline toolkit/tools/imagegen/installutils/installutils.go:1563]: Duplicate code, can we add the sshkey to the list above instead and append all at once?

**Review by romoh — APPROVED** [review]

**Comment by amritakohli** [comment]
add "OS Modifier (EMU):" or something to pr title?

**Review by cwize1 — APPROVED** [review]
Please also get sign-off from @dmcilvaney

**Comment by elainezhao96** [comment]
> Please also get sign-off from @dmcilvaney

oops sorry missed this comment. @dmcilvaney let me know if theres anything needs improve, thanks

**Pull request merged**